src/main/java/org/dsa/iot/knx/KnxIpConnection.java

```diff
@@ -170,61 +185,47 @@ public void handle(ActionResult event) {
 
 	}
 
-	private class Poller implements Runnable {
-		KnxConnection listener;
+	private class NetworkListener implements NetworkLinkListener {
+
+		public void indication(FrameEvent e) {
 
-		public void addListener(KnxConnection listener) {
-			this.listener = listener;
 		}
 
-		@Override
-		public void run() {
-			try {
-				knxLink = new KNXNetworkLinkIP(serviceMode, localEP, remoteEP, useNat, TPSettings.TP1);
-				knxLink.addLinkListener(new NetworkLinkListener() {
+		public void linkClosed(CloseEvent e) {
```

Comment:
Java's convention for methods like these would be `onLinkClosed`, `onConfirmation`, etc..